spec/install/gemfile/platform_spec.rb

```diff
@@ -230,21 +230,46 @@
     expect(out).not_to match(/Could not find gem 'some_gem/)
   end
 
-  it "prints a helpful warning when a dependency is unused on any platform" do
-    simulate_platform "ruby"
-    simulate_ruby_engine "ruby"
+  context "when disable_platform_warnings is false (by default)" do
+    before { bundle! "config disable_platform_warnings false" }
 
-    gemfile <<-G
-      source "file://#{gem_repo1}"
+    it "prints a helpful warning when a dependency is unused on any platform" do
+      simulate_platform "ruby"
+      simulate_ruby_engine "ruby"
 
-      gem "rack", :platform => [:mingw, :mswin, :x64_mingw, :jruby]
-    G
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+
+        gem "rack", :platform => [:mingw, :mswin, :x64_mingw, :jruby]
+      G
 
-    bundle! "install"
+      bundle! "install"
 
-    expect(out).to include <<-O.strip
-The dependency #{Gem::Dependency.new("rack", ">= 0")} will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
-    O
+      expect(out).to include <<-O.strip
+  The dependency #{Gem::Dependency.new("rack", ">= 0")} will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
+      O
+    end
+  end
+
+  context "when disable_platform_warnings is true" do
+    before { bundle! "config disable_platform_warnings true" }
+
+    it "doesnot print the warning when a dependency is unused on any platform" do
+      simulate_platform "ruby"
+      simulate_ruby_engine "ruby"
+
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+
+        gem "rack", :platform => [:mingw, :mswin, :x64_mingw, :jruby]
+      G
+
+      bundle! "install"
+
+      expect(out).not_to include <<-O.strip
+  The dependency #{Gem::Dependency.new("rack", ">= 0")} will be unused by any of the platforms Bundler is installing for. Bundler is installing for ruby but the dependency is only for x86-mingw32, x86-mswin32, x64-mingw32, java. To add those platforms to the bundle, run `bundle lock --add-platform x86-mingw32 x86-mswin32 x64-mingw32 java`.
```

Comment:
This is a lot to match exactly against in the output which could easily be missed if we change the test or behavior. I would match with:

    expect(out).to_not include(/The dependency (.*) will be unused/)

Reply:
Cool. Will change it.

Reply:
Yep, that was a typo on my part.

Reply:
Using 
```bash
expect(out).not_to match(/The dependency (.*) will be unused/)
```
does the trick. Is it good?